ideascube/serveradmin/tests/test_views.py

```diff
@@ -642,3 +642,29 @@ def test_staff_forgets_non_existing_connection(mocker, staffapp):
     assert u'No known Wi-Fi networks' not in res.unicode_body
     assert u'my home network' in res.unicode_body
     assert u'random open network' in res.unicode_body
+
+
+@pytest.mark.usefixtures('installed_packages_catalog')
+def test_staff_should_see_packages_with_cards(staffapp, systemuser):
+    from ideascube.configuration import set_config
+    set_config('ideascube', 'displayed-package-ids', ['test.package1'], systemuser)
+    res = staffapp.get(reverse("server:cards"), status=200)
+    assert 'Test package1' in res.unicode_body
+    assert 'Test package2' in res.unicode_body
+    assert 'Test package3' not in res.unicode_body
+
+    form = res.forms['cards']
+    inputs = form.fields['visible']
+    for _input in inputs:
+        should_be_checked = (_input._value == 'test.package1')
```

Comment:
> If I test id and not value, I don't test an attribute used and I test an attribute not used.

Wait, where did I say you shouldn't test for the value?

Maybe that wasn't clear, so let me expand in excruciating details. :smile: 

Your test code makes several assumptions: that there will be several checkboxes (you're not verifying this assumption), that one of them will have its value set to `'test.package1'` (you're not verifying that assumption either). In addition you, the programmer, make one more assumption: that only one of these checkboxes will have its value set to `'test.package1`'. (you're not verifying it either)

And then your test verifies that the checkbox with that specific value is checked, and that all others aren't.

You decided to make all those assumptions when writing the code. What I'm arguing for is that you could very well decide to make one more (as long as you verify it) : that for each checkbox the id is set accordingly to the value. (for example `id = 'pkg_%s' % value`)

Making this assumption doesn't change at all what your test is verifying (I'm not suggesting you stop checking for the value), and it allows you to stop using a private API from your dependency, so that seems like a net win to me.